### PR TITLE
ci: First run at interchaintests.

### LINKED ISSUE #3212: Testing: add interchaintest powered e2e tests and reproducible docker builds
## Problem

Gaia e2e tests can be improved by adopting interchaintest and creating better scenario test coverage through testing scenarios.

## Closing criteria

Add nightly e2e tests powered by interchaintest.
Build nighly docker images to power test and speed up development.

### PR DESCRIPTION
**Pull request by fastfadingviolets**
## Description

Introduces integration tests using the interchaintest framework.

These include tests for different transaction types, and consumer chain launches. They are triggered by the docker-push workflow and will test the pushed image. They can also be triggered manually and will try to test an image on the branch supplied.

For the gaia team, the main thing to review are the two github workflows.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Provided a link to the relevant issue or specification
- [ ] Reviewed "Files changed" and left comments if necessary 
- [ ] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] Confirmed all author checklist items have been addressed
- [ ] Confirmed that this PR does not change production code

### CODE REVIEW

**Comment by fastfadingviolets** [comment]
You can see these in action in [this fork](https://github.com/fastfadingviolets/gaia/actions/runs/9961220518/job/27522253728)

**Review by MSalopek — COMMENTED** [review]
Can we make this into a standalone module?

This is done by adding a `go.mod` with required dependencies to the `tests/interchain` folder.

This would enable us to version the tests separately from the rest of the repository. As far as I can tell, these tests don't actually use anything from the gaia repo so it should be safe to de-couple them.

File: go.mod [hunk 1]
```diff
@@ -28,7 +31,12 @@ require (
 	github.com/spf13/cobra v1.8.1
 	github.com/spf13/pflag v1.0.5
 	github.com/spf13/viper v1.19.0
+	github.com/strangelove-ventures/interchaintest/v7 v7.0.0-00010101000000-000000000000
```
Comment (MSalopek) [inline go.mod]: IIRC, versions >= v8.4.0 have ICS support built in. We will need to switch to it once we move main to v50 (probably today).

https://github.com/strangelove-ventures/interchaintest/releases/tag/v8.5.0

File: .github/workflows/interchain-test.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            - name: Setup go
+              uses: actions/setup-go@v5
+            - name: Run test
+              env:
+                TEST_DOCKER_REGISTRY: "ghcr.io/${{ github.repository_owner }}"
+                TEST_OLD_GAIA_IMAGE_VERSION: "${{ matrix.previous_version }}"
+                TEST_NEW_GAIA_IMAGE_VERSION: "${{ matrix.test_version }}"
+                TEST_UPGRADE_NAME: "${{ matrix.upgrade_name }}"
+              run: |
+                # This docker pull/tag is a quick hack only necessary for v19, since there were no official v18 images built.
```
Comment (MSalopek) [inline .github/workflows/interchain-test.yml:60]: We can build a v18.1.0 image via manual workflow dispatch.

Comment (fastfadingviolets) [reply .github/workflows/interchain-test.yml:60]: I tried and wasn't able to because the tag doesn't have the workflow

**Comment by fastfadingviolets** [comment]
New set of passing tests is here: https://github.com/fastfadingviolets/gaia/actions/runs/10080366546/job/27870068844

**Review by MSalopek — APPROVED** [review]

**Review by p-offtermatt — APPROVED** [review]
Administrative approval, thx for the work!

**Review by LexaMichaelides — APPROVED** [review]
🤷🏼‍♀️ 👍🏼

**Pull request merged**